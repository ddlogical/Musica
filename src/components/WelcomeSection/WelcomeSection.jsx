import "./WelcomeSection.scss";
import Icon from '../Icon';
import { memo } from 'react';

function WelcomeSection() {

    return (
        <section className='welcome'>

            <div className='container'>

                <div className='welcome__content'>

                    <h1 className='welcome__heading'>Welcome to <span>musica,</span> check out our latest albums</h1>

                    <div className='welcome__proposals'>
                        <div className='welcome__proposal'>
                            <div className='welcome__proposal-heading'>
                                <Icon color='#fff' type='album' />
                                <h3>CHECK OUR CD COLLECTION</h3>
                            </div>
                            <p>Donec pede justo, fringilla vel, al, vulputate  eget, arcu. In </p>
                        </div>
                        <div className='welcome__proposal welcome__proposal--colored'>
                            <div className='welcome__proposal-heading'>
                                <Icon color='#fff' type='headphones' />
                                <h3>LISTEN BEFORE PURCHASE</h3>
                            </div>
                            <p>Donec pede justo, fringilla vel, al, vulputate  eget, arcu. In </p>
                        </div>
                        <div className='welcome__proposal'>
                            <div className='welcome__proposal-heading'>
                                <Icon color='#fff' type='notebook' />
                                <h3>UPCOMING EVENTS</h3>
                            </div>
                            <p>Donec pede justo, fringilla vel, al, vulputate  eget, arcu. In </p>
                        </div>
                    </div>
                </div>

            </div>

        </section>
    );
}

export default memo(WelcomeSection);