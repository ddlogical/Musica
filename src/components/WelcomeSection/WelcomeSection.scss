@import "../../styles/variables.scss";

.welcome {
	position: relative;
	padding: 7.5rem 0 6.2rem 0;

	&__content {
		grid-column: 1 / -1;
		grid-row: 1 / 1;
	}

	&__heading {
		text-align: center;
		margin-bottom: 6rem;

		& span {
			background-color: $color-secondary;
			color: $color-primary-white;
			padding: 0 5px;
		}
	}

	&__proposals {
		display: flex;
		justify-content: space-between;

		@media only screen and (max-width: $bp-medium) {
			flex-direction: column;
			align-items: center;
		}
	}

	&__proposal {
		background-color: $color-primary-dark;
		color: $color-primary-white;
		padding: 1.4rem 2.3rem 1.4rem 1.6rem;
		border-radius: 5px;
		text-decoration: none;
		text-align: center;

		@media only screen and (max-width: $bp-medium) {
			width: 50%;
			margin-bottom: 1rem;
			font-size: 1.5rem;
		}

		@media only screen and (max-width: $bp-small) {
			width: 100%;
		}

		&:hover {
			background-color: $color-secondary;
		}

		&--colored {
			background-color: $color-secondary;
		}
	}

	&__proposal-heading {
		display: flex;
		align-items: center;
		margin-bottom: 1rem;

		@media only screen and (max-width: $bp-medium) {
			justify-content: center;
		}

		& h3 {
			margin-left: 1.5rem;
		}
	}
}
