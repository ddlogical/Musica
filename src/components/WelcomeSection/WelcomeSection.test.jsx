import { render } from "@testing-library/react";

import WelcomeSection from "./";

test("WelcomeSection snapshot", () => {

    const view = render(<WelcomeSection />);

    expect(view).toMatchSnapshot();
})