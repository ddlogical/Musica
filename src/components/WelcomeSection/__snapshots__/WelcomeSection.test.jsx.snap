// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeSection snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="welcome"
      >
        <div
          class="container"
        >
          <div
            class="welcome__content"
          >
            <h1
              class="welcome__heading"
            >
              Welcome to 
              <span>
                musica,
              </span>
               check out our latest albums
            </h1>
            <div
              class="welcome__proposals"
            >
              <div
                class="welcome__proposal"
              >
                <div
                  class="welcome__proposal-heading"
                >
                  <svg
                    fill="#fff"
                    height="48"
                    width="48"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M24 32.2q3.5 0 6-2.375T32.5 24q0-3.55-2.475-6.025Q27.55 15.5 24 15.5q-3.45 0-5.825 2.5T15.8 24q0 3.45 2.375 5.825T24 32.2Zm0-6.2q-.85 0-1.425-.575Q22 24.85 22 24q0-.85.575-1.425Q23.15 22 24 22q.85 0 1.425.575Q26 23.15 26 24q0 .85-.575 1.425Q24.85 26 24 26Zm0 18q-4.1 0-7.75-1.575-3.65-1.575-6.375-4.3-2.725-2.725-4.3-6.375Q4 28.1 4 24q0-4.15 1.575-7.8 1.575-3.65 4.3-6.35 2.725-2.7 6.375-4.275Q19.9 4 24 4q4.15 0 7.8 1.575 3.65 1.575 6.35 4.275 2.7 2.7 4.275 6.35Q44 19.85 44 24q0 4.1-1.575 7.75-1.575 3.65-4.275 6.375t-6.35 4.3Q28.15 44 24 44Zm0-3q7.1 0 12.05-4.975Q41 31.05 41 24q0-7.1-4.95-12.05Q31.1 7 24 7q-7.05 0-12.025 4.95Q7 16.9 7 24q0 7.05 4.975 12.025Q16.95 41 24 41Zm0-17Z"
                    />
                  </svg>
                  <h3>
                    CHECK OUR CD COLLECTION
                  </h3>
                </div>
                <p>
                  Donec pede justo, fringilla vel, al, vulputate  eget, arcu. In 
                </p>
              </div>
              <div
                class="welcome__proposal welcome__proposal--colored"
              >
                <div
                  class="welcome__proposal-heading"
                >
                  <svg
                    fill="#fff"
                    height="48"
                    width="48"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M16.4 42H9q-1.2 0-2.1-.9Q6 40.2 6 39V24q0-3.75 1.425-7.025 1.425-3.275 3.85-5.7 2.425-2.425 5.7-3.85Q20.25 6 24 6q3.75 0 7.025 1.425 3.275 1.425 5.7 3.85 2.425 2.425 3.85 5.7Q42 20.25 42 24v15q0 1.2-.9 2.1-.9.9-2.1.9h-7.4V27.2H39V24q0-6.25-4.375-10.625T24 9q-6.25 0-10.625 4.375T9 24v3.2h7.4Zm-3-11.8H9V39h4.4Zm21.2 0V39H39v-8.8Zm0 0H39h-4.4Zm-21.2 0H9Z"
                    />
                  </svg>
                  <h3>
                    LISTEN BEFORE PURCHASE
                  </h3>
                </div>
                <p>
                  Donec pede justo, fringilla vel, al, vulputate  eget, arcu. In 
                </p>
              </div>
              <div
                class="welcome__proposal"
              >
                <div
                  class="welcome__proposal-heading"
                >
                  <svg
                    fill="#fff"
                    height="48"
                    width="48"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m21.75 35.75-6.2-6.15 1.5-1.5 4.7 4.6 8.95-9 1.55 1.55Zm-11.8 7.1q-1.15 0-1.975-.85-.825-.85-.825-1.95V11.2q0-1.15.825-1.975Q8.8 8.4 9.95 8.4h3.75V4.9h2.45v3.5H31.9V4.9h2.4v3.5h3.75q1.15 0 1.975.825.825.825.825 1.975v28.85q0 1.1-.825 1.95-.825.85-1.975.85Zm0-2.2h28.1q.25 0 .425-.2t.175-.4V20.3H9.35v19.75q0 .2.175.4t.425.2Zm-.6-22.55h29.3v-6.9q0-.25-.175-.425t-.425-.175H9.95q-.25 0-.425.175t-.175.425Zm0 0v-7.5 7.5Z"
                    />
                  </svg>
                  <h3>
                    UPCOMING EVENTS
                  </h3>
                </div>
                <p>
                  Donec pede justo, fringilla vel, al, vulputate  eget, arcu. In 
                </p>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="welcome"
    >
      <div
        class="container"
      >
        <div
          class="welcome__content"
        >
          <h1
            class="welcome__heading"
          >
            Welcome to 
            <span>
              musica,
            </span>
             check out our latest albums
          </h1>
          <div
            class="welcome__proposals"
          >
            <div
              class="welcome__proposal"
            >
              <div
                class="welcome__proposal-heading"
              >
                <svg
                  fill="#fff"
                  height="48"
                  width="48"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M24 32.2q3.5 0 6-2.375T32.5 24q0-3.55-2.475-6.025Q27.55 15.5 24 15.5q-3.45 0-5.825 2.5T15.8 24q0 3.45 2.375 5.825T24 32.2Zm0-6.2q-.85 0-1.425-.575Q22 24.85 22 24q0-.85.575-1.425Q23.15 22 24 22q.85 0 1.425.575Q26 23.15 26 24q0 .85-.575 1.425Q24.85 26 24 26Zm0 18q-4.1 0-7.75-1.575-3.65-1.575-6.375-4.3-2.725-2.725-4.3-6.375Q4 28.1 4 24q0-4.15 1.575-7.8 1.575-3.65 4.3-6.35 2.725-2.7 6.375-4.275Q19.9 4 24 4q4.15 0 7.8 1.575 3.65 1.575 6.35 4.275 2.7 2.7 4.275 6.35Q44 19.85 44 24q0 4.1-1.575 7.75-1.575 3.65-4.275 6.375t-6.35 4.3Q28.15 44 24 44Zm0-3q7.1 0 12.05-4.975Q41 31.05 41 24q0-7.1-4.95-12.05Q31.1 7 24 7q-7.05 0-12.025 4.95Q7 16.9 7 24q0 7.05 4.975 12.025Q16.95 41 24 41Zm0-17Z"
                  />
                </svg>
                <h3>
                  CHECK OUR CD COLLECTION
                </h3>
              </div>
              <p>
                Donec pede justo, fringilla vel, al, vulputate  eget, arcu. In 
              </p>
            </div>
            <div
              class="welcome__proposal welcome__proposal--colored"
            >
              <div
                class="welcome__proposal-heading"
              >
                <svg
                  fill="#fff"
                  height="48"
                  width="48"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16.4 42H9q-1.2 0-2.1-.9Q6 40.2 6 39V24q0-3.75 1.425-7.025 1.425-3.275 3.85-5.7 2.425-2.425 5.7-3.85Q20.25 6 24 6q3.75 0 7.025 1.425 3.275 1.425 5.7 3.85 2.425 2.425 3.85 5.7Q42 20.25 42 24v15q0 1.2-.9 2.1-.9.9-2.1.9h-7.4V27.2H39V24q0-6.25-4.375-10.625T24 9q-6.25 0-10.625 4.375T9 24v3.2h7.4Zm-3-11.8H9V39h4.4Zm21.2 0V39H39v-8.8Zm0 0H39h-4.4Zm-21.2 0H9Z"
                  />
                </svg>
                <h3>
                  LISTEN BEFORE PURCHASE
                </h3>
              </div>
              <p>
                Donec pede justo, fringilla vel, al, vulputate  eget, arcu. In 
              </p>
            </div>
            <div
              class="welcome__proposal"
            >
              <div
                class="welcome__proposal-heading"
              >
                <svg
                  fill="#fff"
                  height="48"
                  width="48"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m21.75 35.75-6.2-6.15 1.5-1.5 4.7 4.6 8.95-9 1.55 1.55Zm-11.8 7.1q-1.15 0-1.975-.85-.825-.85-.825-1.95V11.2q0-1.15.825-1.975Q8.8 8.4 9.95 8.4h3.75V4.9h2.45v3.5H31.9V4.9h2.4v3.5h3.75q1.15 0 1.975.825.825.825.825 1.975v28.85q0 1.1-.825 1.95-.825.85-1.975.85Zm0-2.2h28.1q.25 0 .425-.2t.175-.4V20.3H9.35v19.75q0 .2.175.4t.425.2Zm-.6-22.55h29.3v-6.9q0-.25-.175-.425t-.425-.175H9.95q-.25 0-.425.175t-.175.425Zm0 0v-7.5 7.5Z"
                  />
                </svg>
                <h3>
                  UPCOMING EVENTS
                </h3>
              </div>
              <p>
                Donec pede justo, fringilla vel, al, vulputate  eget, arcu. In 
              </p>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
